### PR TITLE
Fix modify missing in some event handlers

### LINKED ISSUE #279: Modify instance missing in the latest omnichannel event handlers
I'm working on a new App which will need to listen to the latest Omnichannel event handlers implemented,
such as `executePostLivechatRoomStarted`, `executePostLivechatRoomClosed`, `executePostLivechatAgentAssigned` and so on.
The fact is that those event handlers don't provide the `modify` instance as a parameter as the `executePostMessageSent` does, so I just can move forward since I need, for instance, send messages to Omnichannel rooms and I need the `modifier` to do it.

Is there a way to add `modify` to these events quickly? I'm asking because the new App I'm working on is a high-priority demand and our plan is to deliver it ASAP.

### PR DESCRIPTION
**Pull request by lolimay**
# What? :boat:

Some Livechat event handlers lack of `modify` to create messages etc. Create this PR to fix the issue.

# Why? :thinking:

Closes #279 

# Links :earth_americas:

# PS :eyes:

### CODE REVIEW

**Comment by lolimay** [comment]
The minimal app for testing the functionality:

```typescript
import { IAppAccessors, IHttp, ILogger, IModify, IPersistence, IRead } from '@rocket.chat/apps-engine/definition/accessors';
import { App } from '@rocket.chat/apps-engine/definition/App';
import {
    ILivechatEventContext, ILivechatRoom, ILivechatRoomClosedHandler, IPostLivechatAgentAssigned, IPostLivechatAgentUnassigned, IPostLivechatRoomClosed,
    IPostLivechatRoomStarted,
} from '@rocket.chat/apps-engine/definition/livechat';
import { IAppInfo } from '@rocket.chat/apps-engine/definition/metadata';

export class LivechatTestingApp extends App implements IPostLivechatAgentAssigned, IPostLivechatAgentUnassigned, IPostLivechatRoomStarted, IPostLivechatRoomClosed {
    constructor(info: IAppInfo, logger: ILogger, accessors: IAppAccessors) {
        super(info, logger, accessors);
    }

    public async executePostLivechatAgentAssigned(data: ILivechatEventContext, read: IRead, http: IHttp, modify: IModify, persis: IPersistence) {
        console.log('executePostLivechatAgentAssigned', { data, read, http, modify, persis });
    }

    public async executePostLivechatAgentUnassigned(data: ILivechatEventContext, read: IRead, http: IHttp, modify: IModify, persis: IPersistence) {
        console.log('executePostLivechatAgentUnassigned', { data, read, http, modify, persis });
    }

    public async executePostLivechatRoomStarted(data: ILivechatRoom, read: IRead, http: IHttp, modify: IModify, persis: IPersistence) {
        console.log('executePostLivechatRoomStarted', { data, read, http, modify, persis });
    }

    public async executePostLivechatRoomClosed(data: ILivechatRoom, read: IRead, http: IHttp, modify: IModify, persis: IPersistence) {
        console.log('executePostLivechatRoomClosed', { data, read, http, modify, persis });
    }
}
```

Zip:  [livechat-testing_0.0.1.zip](https://github.com/RocketChat/Rocket.Chat.Apps-engine/files/4870164/livechat-testing_0.0.1.zip)

**Comment by renatobecker** [comment]
Thanks @lolimay for the fast work on this demand.
We really need those fixes in order to implement a new and very important Omnichannel App.

**Comment by renatobecker** [comment]
@lolimay another important thing to mention:
I just noticed that, for some reason, the `LivechatRoomClosedHandler` is being triggered twice, which is raising other errors on my App.

Can you check it, also?

**Comment by lolimay** [comment]
> @lolimay another important thing to mention:
> I just noticed that, for some reason, the `LivechatRoomClosedHandler` is being triggered twice, which is raising other errors on my App.
> 
> Can you check it, also?

Fixed it https://github.com/RocketChat/Rocket.Chat.Apps-engine/pull/280/commits/4877047f62379a26bcfefff08aa7ff390554b2a3

**Comment by renatobecker** [comment]
@d-gubert any chance you get this fix merged and released soon?
I'm getting stuck on the new app I'm implementing and there is a deadline to deliver it, so I'm wondering if we could release this fix out as a hotfix release.
Is it possible?

File: src/definition/livechat/IPostLivechatAgentAssigned.ts [hunk 1]
```diff
@@ -12,7 +12,10 @@ export interface IPostLivechatAgentAssigned {
      * @param data the livechat context data which contains agent's info and room's info.
      * @param read An accessor to the environment
      * @param http An accessor to the outside world
-     * @param persistence An accessor to the App's persistence
+     * @param persis An accessor to the App's persistence
+     * @param modify An accessor to the modifier
      */
-    [AppMethod.EXECUTE_POST_LIVECHAT_AGENT_ASSIGNED](context: ILivechatEventContext, read: IRead, http: IHttp, persistence: IPersistence): Promise<void>;
+    [AppMethod.EXECUTE_POST_LIVECHAT_AGENT_ASSIGNED](
+        context: ILivechatEventContext, read?: IRead, http?: IHttp, persis?: IPersistence, modify?: IModify,
```
Comment (d-gubert) [inline src/definition/livechat/IPostLivechatAgentAssigned.ts:19]:
```suggestion
        context: ILivechatEventContext, read: IRead, http: IHttp, persis: IPersistence, modify?: IModify,
```

There is no need to mark the existing parameters as optional, only the new one

File: src/definition/livechat/IPostLivechatAgentUnassigned.ts [hunk 2]
```diff
@@ -12,7 +12,10 @@ export interface IPostLivechatAgentUnassigned {
      * @param data the livechat context data which contains agent's info and room's info.
      * @param read An accessor to the environment
      * @param http An accessor to the outside world
-     * @param persistence An accessor to the App's persistence
+     * @param persis An accessor to the App's persistence
+     * @param modify An accessor to the modifier
      */
-    [AppMethod.EXECUTE_POST_LIVECHAT_AGENT_UNASSIGNED](context: ILivechatEventContext, read: IRead, http: IHttp, persistence: IPersistence): Promise<void>;
+    [AppMethod.EXECUTE_POST_LIVECHAT_AGENT_UNASSIGNED](
+        context: ILivechatEventContext, read?: IRead, http?: IHttp, persis?: IPersistence, modify?: IModify,
```
Comment (d-gubert) [inline src/definition/livechat/IPostLivechatAgentUnassigned.ts:19]:
```suggestion
        context: ILivechatEventContext, read: IRead, http: IHttp, persis: IPersistence, modify?: IModify,
```

There is no need to mark the existing parameters as optional, only the new one

File: src/definition/livechat/IPostLivechatRoomClosed.ts [hunk 3]
```diff
@@ -12,7 +12,10 @@ export interface IPostLivechatRoomClosed {
      * @param livechatRoom The livechat room which is closed.
      * @param read An accessor to the environment
      * @param http An accessor to the outside world
-     * @param persistence An accessor to the App's persistence
+     * @param persis An accessor to the App's persistence
+     * @param modify An accessor to the modifier
      */
-    [AppMethod.EXECUTE_POST_LIVECHAT_ROOM_CLOSED](room: ILivechatRoom, read: IRead, http: IHttp, persistence: IPersistence): Promise<void>;
+    [AppMethod.EXECUTE_POST_LIVECHAT_ROOM_CLOSED](
+        room: ILivechatRoom, read?: IRead, http?: IHttp, persis?: IPersistence, modify?: IModify,
```
Comment (d-gubert) [inline src/definition/livechat/IPostLivechatRoomClosed.ts:19]:
```suggestion
        room: ILivechatRoom, read: IRead, http: IHttp, persis: IPersistence, modify?: IModify,
```

There is no need to mark the existing parameters as optional, only the new one

File: src/definition/livechat/IPostLivechatRoomStarted.ts [hunk 4]
```diff
@@ -12,7 +12,10 @@ export interface IPostLivechatRoomStarted {
      * @param livechatRoom The livechat room which is started.
      * @param read An accessor to the environment
      * @param http An accessor to the outside world
-     * @param persistence An accessor to the App's persistence
+     * @param persis An accessor to the App's persistence
+     * @param modify An accessor to the modifier
      */
-    [AppMethod.EXECUTE_POST_LIVECHAT_ROOM_STARTED](room: ILivechatRoom, read: IRead, http: IHttp, persistence: IPersistence): Promise<void>;
+    [AppMethod.EXECUTE_POST_LIVECHAT_ROOM_STARTED](
+        room: ILivechatRoom, read?: IRead, http?: IHttp, persis?: IPersistence, modify?: IModify,
```
Comment (d-gubert) [inline src/definition/livechat/IPostLivechatRoomStarted.ts:19]:
```suggestion
        room: ILivechatRoom, read: IRead, http: IHttp, persis: IPersistence, modify?: IModify,
```

There is no need to mark the existing parameters as optional, only the new one

**Pull request merged**